dlib/matrix/matrix_fft.h

```diff
@@ -263,6 +264,291 @@ namespace dlib
+            matrix<R> win(1, wlen);
+            for (std::size_t i = 0 ; i < wlen ; ++i)
+                win(0, i) = w(i, wlen);
+            matrix<R> win2 = squared(win);
+
+            for (long t = 0 ; t < stft.nr() ; ++t)
+            {
+                set_subm(signal, 0, t*hoplen, 1, wlen) += pointwise_multiply(win, subm(ifft_obj(rowm(stft, t)), 0, 0, 1, wlen));
+                set_subm(norm,   0, t*hoplen, 1, wlen) += win2;
+            }
+
+            /*! Remove padding of wlen/2 and wlen/2 on either end !*/
+            DLIB_ASSERT(sum(subm(norm, 0, wlen/2, 1, ntime - wlen) < 1e-13) == 0, "NOLA constraint not satisfied");
+            signal = pointwise_divide(subm(signal, 0, wlen/2, 1, ntime - wlen),
+                                      subm(norm,   0, wlen/2, 1, ntime - wlen));
+
+            return signal;
+        }
+    }
+
+// ----------------------------------------------------------------------------------------
+
+    const auto hann_window = [](std::size_t i, std::size_t N) 
+    /*!
+        WHAT THIS OBJECT REPRESENTS
+            This returns a function object with signature double(size_t i, size_t N) which computes
+            a periodic Hann window which can be passed to the STFT family of functions.
+    !*/
+    {
+        return hann(i, N, PERIODIC);
+    };
+
+    const auto blackman_window = [](std::size_t i, std::size_t N) 
+    /*!
+        WHAT THIS OBJECT REPRESENTS
+            This returns a function object with signature double(size_t i, size_t N) which computes
+            a periodic Blackman window which can be passed to the STFT family of functions.
+    !*/
+    {
+        return blackman(i, N, PERIODIC);
+    };
+    
+    const auto blackman_nuttall_window = [](std::size_t i, std::size_t N) 
+    /*!
+        WHAT THIS OBJECT REPRESENTS
+            This returns a function object with signature double(size_t i, size_t N) which computes
+            a periodic Blackman-Nuttall window which can be passed to the STFT family of functions.
+    !*/
+    {
+        return blackman_nuttall(i, N, PERIODIC);
+    };
+
+    const auto blackman_harris_window = [](std::size_t i, std::size_t N) 
+    /*!
+        WHAT THIS OBJECT REPRESENTS
+            This returns a function object with signature double(size_t i, size_t N) which computes
+            a periodic Blackman-Harris window which can be passed to the STFT family of functions.
+    !*/
+    {
+        return blackman_harris(i, N, PERIODIC);
+    };
+
+    const auto blackman_harris7_window = [](std::size_t i, std::size_t N) 
+    /*!
+        WHAT THIS OBJECT REPRESENTS
+            This returns a function object with signature double(size_t i, size_t N) which computes
+            a periodic 7th order Blackman-Harris window which can be passed to the STFT family of functions.
+    !*/
+    {
+        return blackman_harris7(i, N, PERIODIC);
+    };
+
+    inline auto kaiser_window(beta_t beta)
```

Comment:
I don't like this one because it makes it look like it's doing something different to the other functions. They are all doing the same thing, it just so happens kaiser requires an extra parameter, and others in the future could too, e.g. gauss window

Reply:
Could you have another think about this? In my opinion, they should all be tag types, not publicly callable things. And I'm not a fan of having some of them having different names that make them sound like factory functions, even if they are... That should be hidden to the user ideally.

Reply:
Alternatively they are all called `make_xxx_window()`, most of them taking no args, except for kaiser which takes beta. Then the name is clear, the function is returning another function.

Reply:
Yeah lets go with that.

Reply:
Yeah that's a good way :)